pkg/ddevapp/backdrop.go

```diff
@@ -62,50 +88,102 @@ ini_set('session.gc_maxlifetime', 200000);
 ini_set('session.cookie_lifetime', 2000000);
 `
 
-// createBackdropSettingsFile creates the app's settings.php or equivalent,
-// adding things like database host, name, and password.
-// Returns the full path to the settings file and err.
+// createBackdropSettingsFile manages creation and modification of settings.php and settings.ddev.php.
+// If a settings.php file already exists, it will be modified to ensure that it includes
+// settings.ddev.php, which contains ddev-specific configuration.
 func createBackdropSettingsFile(app *DdevApp) (string, error) {
-	settingsFilePath, err := app.DetermineSettingsPathLocation()
+	settings := NewBackdropSettings()
+
+	if !fileutil.FileExists(app.SiteSettingsPath) {
```

Comment:
I agree -- this is checking existence of the "main" settings.php file and writing that if one doesn't exist.

Reply:
I was saying that perhaps it should check #ddev-generated and if it *is* always recreate. That lets us future-proof. Although I certainly understand that if we were making changes we would put them into the settings.ddev.php.

Reply:
If I understand this correctly, I think this means we'd need two signatures (as @rickmanelius pointed out in his review) because this currently writes the `#ddev-generated` signature to the appended portion of settings.php, so it would nearly always be present in the file.  I'm not against that at all, as it does clearly separate ownership vs. modification.

Reply:
Yeah, we should *not* be writing any signature when *appending* to a user settings.php, IMO.